File: app/src/main/java/dev/test/trustisend/entity/ActiveFile.java
package dev.test.trustisend.entity;

public class ActiveFile extends Group {
    private final String path;
    private String fileUUID;
    private final FileScanStatus scanStatus;
    public ActiveFile(String groupUUID, String userEmail, String timestamp, Integer numberDownloads, String fileUUID, String path, FileScanStatus scanStatus) {
        super(groupUUID, userEmail, timestamp, numberDownloads);
        this.fileUUID = fileUUID;
        this.path = path;
        this.scanStatus = scanStatus;
    }
    public ActiveFile(Group group, String fileUUID, String path, FileScanStatus scanStatus) {
        super(group.getGroupUUID(), group.getUserEmail(), group.getTimestamp(), group.getNumberDownloads());
        this.fileUUID = fileUUID;
        this.path = path;
        this.scanStatus = scanStatus;
    }
    public ActiveFile(Group group, String path, FileScanStatus scanStatus) {
        super(group.getGroupUUID(), group.getUserEmail(), group.getTimestamp(), group.getNumberDownloads());
        this.path = path;
        this.scanStatus = scanStatus;
    }

    public String getFileUUID() {
        return fileUUID;
    }
    public String getPath() {
        return path;
    }
    public FileScanStatus getScanStatus() {return scanStatus;}
}

File: dev/FileScanner/activefile.py
import group as Group

class ActiveFile():
    def __init__(self, group_uuid, file_uuid, path, scan_status):
        self.group_uuid = group_uuid
        self.file_uuid = file_uuid
        self.path = path
        self.scan_status = scan_status

    # Getter methods
    def get_file_uuid(self):
        return self.file_uuid

    def get_path(self):
        return self.path

    def get_scan_status(self):
        return self.scan_status